cognite/extractorutils/rest/extractor.py

```diff
@@ -146,13 +146,57 @@ def __init__(
             config_class=config_class,
             config_file_path=config_file_path,
         )
-        self._default_base_url = base_url or ""
+        self._default_base_url = default_base_url or ""
         self.headers: Dict[str, Union[str, Callable[[], str]]] = headers or {}
-        self.endpoints: List[Endpoint] = []
+        self.endpoints: Optional[List[Endpoint]] = []
```

Comment:
I don't particularly like this way of using `None` to indicate a running state, it needs to at least be commented about because it's a very particular way of doing it. I don't immediately see a better solution, though.

But I think we need two changes, both to add a comment here explaining what a `None` for endpoints mean conceptually, and make it 'private' so that outside users are not messing around with it expecting it to work a certain way

Reply:
We could just keep a boolean, like `_is_extractor_started`, but we should still clear the `endpoints` list after writing it into the queue, so I'm not sure if it's really better.

Reply:
I renamed it to `_initial_endpoints`, which I think is a bit clearer, and also added a comment.